src/databricks/labs/ucx/source_code/known.py

```diff
@@ -56,13 +56,22 @@ def __init__(self):
             self._module_problems[name] = []
 
     @staticmethod
-    def _get_known():
-        module = __name__
-        if __name__ == "__main__":  # code path for UCX developers invoking `make known`
-            module = "databricks.labs.ucx.source_code.known"
-        # load known.json from package data, because we may want to use zipapp packaging
-        data = pkgutil.get_data(module, "known.json")
-        return json.loads(data)
+    def _get_known() -> dict[str, dict[str, list[dict[str, str]]]]:
+        """Loads the set of known distributions (and false positives), included with this module.
+
+        Returns:
+            A dictionary (keyed by distribution name) containing the known problems for each module.
+        Raises:
+            FileNotFoundError: If the builtin set of known distribution is missing.
+        """
+        this_module = __package__
+        if this_module is None:
+            # When this file is invoked directly as a script, __package__ is None.
+            # Note: can get confused if this package is _also_ installed in site-packages.
+            from databricks.labs.ucx import source_code as this_module  # pylint: disable=import-outside-toplevel
+        known_json = importlib.resources.files(this_module) / "known.json"
```

Comment:
why do we need this change? does it work with zipapps? does it work with PEX setup? `importlib.resources` had issues when invoked from zip archives - we don't know yet if UCX will be deployed to Databricks runtime as ZIP or not.

Reply:
We don't need this change; `pkgutil` is just something I try to avoid, partly because it's deprecated, partly because in ancient times I also had issues it.

Given your reservations and complexity of testing I'll revert this.